src/gc/gc.cpp

```diff
@@ -12132,7 +12133,10 @@ void gc_heap::bgc_loh_alloc_clr (uint8_t* alloc_start,
         }
     }
 #endif //VERIFY_HEAP
-    
+
+    // TODO: VS are we actually holding the right lock? check  loh vs. soh heap locks
```

Comment:
Don't checkin a todo comment like this.

Reply:
I agree with David - this comment should definitely not be checked and as I mentioned above you should be splitting the total_alloc_bytes because there's a separate lock for SOH and LOH. even if you don't repro this in the test, that'd only be because the test is not hitting this scenario. at this point in the code it's only holding the LOH's lock. another thread could be allocating on SOH and modifying the total_alloc_bytes at the same time

Reply:
Yes, we need two counts here because there are two locks. TODO tracks that.

I am trying to have a test that would be sensitive to this. We evidently do not have one, but it is tricky. The heap affinitization make us "lucky" most of the time, while what we have is clearly unsafe.
I will probably just have something that mixes SOH/LOH/Threading.

Reply:
Any thoughts on the tearing issue on 32bit? 
I think we should just permit/ignore tearing. 

Note - Per-heap values have the same issue too and going to the lengths needed to prevent that in the context of "dirty" API seems a bit paranoid.

Reply:
sorry I missed this one - I don't have a problem with not using interlock; I agree with your change in the impl of `GCInterface::GetTotalAllocatedBytes`.